app/models/concerns/host_params.rb

```diff
@@ -0,0 +1,76 @@
+module HostParams
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :host_parameters, :dependent => :destroy, :foreign_key => :reference_id, :inverse_of => :host
+    has_many :parameters, :dependent => :destroy, :foreign_key => :reference_id, :class_name => "HostParameter"
+    accepts_nested_attributes_for :host_parameters, :allow_destroy => true
+    include ParameterValidators
+    attr_reader :cached_host_params
+
+    def params
+      host_params.update(lookup_keys_params)
+    end
+
+    def clear_host_parameters_cache!
+      @cached_host_params = nil
+    end
+
+    def host_inherited_params(include_source = false)
+      hp = {}
+      params = host_inherited_params_objects
+      params.each do |param|
+        case param
+          when CommonParameter
+            hp.update(Hash[param.name => include_source ? {:value => param.value, :source => N_('common').to_sym, :safe_value => param.safe_value} : param.value])
+          when GroupParameter
+            hp.update(Hash[param.name => include_source ? {:value => param.value, :source => N_('hostgroup').to_sym, :safe_value => param.safe_value, :source_name => hostgroup.title} : param.value])
```

Comment:
Sorry, I can't see why this parameter is not in the group below? It does the same update call doesn't it?

`hp.update(Hash[param.name => include_source ? {:value => param.value, :source => source.to_sym, :safe_value => param.safe_value, :source_name => hostgroup.title} : param.value])`